src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/CallInstruction.cs

```diff
@@ -281,6 +282,10 @@ internal class MethodInfoCallInstruction : CallInstruction
         protected readonly MethodInfo _target;
         protected readonly int _argumentCount;
 
+        private const int CacheLength = 16;
+        [ThreadStatic]
```

Comment:
Why not use `ArrayPool<object>.Shared`?

Reply:
Pool produces arrays that are “at least this long”, but we need exact length here.